problem-gson/src/main/java/org/zalando/problem/TypeURIAdapter.java

```diff
@@ -0,0 +1,41 @@
+package org.zalando.problem;
+
+import com.google.gson.JsonIOException;
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonToken;
+import com.google.gson.stream.JsonWriter;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+final class TypeURIAdapter extends TypeAdapter<URI> {
+
+    public static final TypeAdapter<URI> INSTANCE = new TypeURIAdapter();
+
+    private TypeURIAdapter() {
+        // Singleton.
+    }
+
+    @Override
+    public void write(JsonWriter out, @Nullable URI value) throws IOException {
+        out.value(value == null || value.equals(Problem.DEFAULT_TYPE) ? null : value.toASCIIString());
+    }
+
+    @Override
+    public URI read(JsonReader in) throws IOException {
+        if (in.peek() == JsonToken.NULL) {
+            in.nextNull();
+            return Problem.DEFAULT_TYPE;
+        }
+        try {
+            String nextString = in.nextString();
+            return "null".equals(nextString) ? Problem.DEFAULT_TYPE : new URI(nextString);
```

Comment:
What case is this?

This is already covered by checking for `JsonToken.NULL`, correct?
```json
{"type":null}
```

Is this in case of:

```json
{"type":"null"}
```

If yes, is that a use case?!

Reply:
@whiskeysierra Yes it's for `{"type":"null"}`. But I'm not sure if it's a valid use case.
I feel we can let this be here if Jackson also supports similar behavior.

Reply:
I don't think it does. I'd like to remove it, since `null` is very different from `"null"`.

Reply:
Okay, I'll remove it.